File: w3-stack/cpluscplus/Stack.cpp
#ifndef STACK
#define STACK

template <class T>
class Stack
{
private:
    void operator=(const Stack &) {}
    Stack(const Stack &) {}

public:
    Stack() {}
    virtual ~Stack() {}
    virtual void clear() = 0;
    virtual void push(T) = 0;
    virtual T pop() = 0;
    virtual long long length() = 0;
    virtual T top() = 0;
    virtual void setDirection(int) {} // only Array base implementation will override setDirection as this is not relevant for LinkList based implementatin.
};

#endif

File: w3-stack/pythonic/Stack.py
from abc import ABC, abstractmethod

class Stack(ABC):
    @abstractmethod
    def clear(self):
        pass
    @abstractmethod
    def length(self):
        pass
    @abstractmethod
    def push(self, value):
        pass
    @abstractmethod
    def pop(self):
        pass
    @abstractmethod
    def top(self):
        pass